File: submissions/nightmare-builders/9.java
	static void f(String storyboard) {
		int balance = 0;
		int lifes = 3;
		int invulnerability = 0;
		boolean superM = false;
		String[] steps = storyboard.split(" +");
		for (int i = 0; i < steps.length; i++) {
			String step = steps[i];
			if (step.matches("\\d+")) {
				balance+=Integer.parseInt(step);
				while (balance > 100) {
					balance-=100;
					lifes++;
				}
					
			}
			else if (step.matches("goomba|koopa|piranha")) {
				if (invulnerability == 0) {
					if (superM) {
						superM = false;
					}
					else {
						lifes--;
					}
				}

			}
			else if ("Mushroom".equals(step)) {
				superM = true;
			}
			else if ("Star".equals(step)) {
				invulnerability = 3;
			}
			else if ("1Up".equals(step)) {
				lifes++;
			}
			else if (step.matches("Bowser|Princess") ) {
				finish("WIN");
			}
			if (lifes == 0) {
				finish("GAME OVER");
			}
			invulnerability = invulnerability > 0 ? --invulnerability : invulnerability;
		}
		finish("???");
	}
	
	static void finish(String s) {
		System.out.println(s);
		System.exit(0);
	}

File: submissions/nightmare-builders/9.py
import re
def f(s):
	b=0;l=3;i=0;m=False;t=s.split(" ")
	for u in t:
		if u.isdigit():
			b+=int(u)
			while b>100:b-=100;l+=1
		elif re.match("(g|k|p).*",u):
			if(i==0):
				if(m):m=False
				else:l-=1
		elif"Mushroom"==u:m=True
		elif"Star"==u:i=3
		elif"1Up"==u:l+=1
		elif re.match("(B|P).*",u):print("WIN");return;
		if(l==0):print("GAME OVER");return;
		i=i-1 if i>0 else i
	print("???");return;